docs/Course Wiki/ECE Course Offerings/ECE407.md

```diff
@@ -0,0 +1,55 @@
+# ECE407
+
+ECE407 (Cryptography) is a 3 credit hour course from both the CS and ECE departments. It is offered in both the fall and spring semesters.
+
+## Content Covered
+### Symmetric Key Cryptography
+- Provable Security
+- One-Time Pad
+- Pseudorandom Functions and Pseudorandom Generators
+- Block Ciphers
+- Message Authentication Codes
+- CPA and CCA Security
+- Cryptographic Hash Functions
+### Public Key Cryptography
+- Hardness Assumptions
+- Basic Group Theory
+- Key Exchange
+- Digital Signatures
+- RSA Cryptosystem
+- Random Oracle Model
+### MPC and ZKP
+- Zero Knowledge Proofs
+- Multi Party Computation
+- Garbled Circuits
+- Secret Shares
+  
+ECE407 is meant to be an undergraduate introduction to modern cryptography. It first covers symmetric key cryptography, which answers the question "How do two parties communicate securely?". This includes topics like one-time pads, perfect secrecy, and PRFs and PRGs. Next it covers public-key crytography, which expands on symmetric-key cryptography but allows for multiple parties to communicate with one. This covers topics like the RSA cryptosystem and key exchange protocls. There is very interesting math used to construct public-key schemes. Finally, the course covers Multi-Party Computation and Zero-Knowledge Proofs. Which answer the questions "How do I prove that I know the answer to a problem without showing the answer?" and "How do two parties jointly compute a function over inputs without revealing what those inputs are to each other?". 
+
+## Prerequisites
+- [CS225](../CS%20Course%20Offerings/CS225.md)
+- [ECE374](ECE374.md)
+
+CS225 and ECE374 are the only official prequisites for this course. CS225 is useful because most homeworks have a small coding portion that requires students to know how to write C++. ECE374 is required because it is important to be able to analyze the runtimes of algorithms, as well as understand reductions and NP-(hardness/completeness). [ECE313](ECE313.md) may also be helpful although not necessary, as security definitions are often written in terms of probabilities. It is doable to take 407 at the same time as 374, as 374 only helps understand the formal definitions a bit more, however it is not neccessary to do well in the course. 
```

Comment:
Link the CS225 and ECE374 pages the first time they are mentioned in the paragraph, as well.  Also, the ECE374 page link doesn't work.  Change to "[ECE313](ECE313.md) may also be helpful, as security definitions are often written in terms of probabilities, although it is not necessary." "Necessary" is misspelled in the last sentence.